src/library/scala/math/BigInt.scala

```diff
@@ -132,11 +212,13 @@ final class BigInt(val bigInteger: BigInteger)
     case that: Float      => isValidFloat && toFloat == that
     case x                => isValidLong && unifiedPrimitiveEquals(x)
   }
-  override def isValidByte: Boolean = this >= Byte.MinValue && this <= Byte.MaxValue
-  override def isValidShort: Boolean = this >= Short.MinValue && this <= Short.MaxValue
-  override def isValidChar: Boolean = this >= Char.MinValue && this <= Char.MaxValue
-  override def isValidInt: Boolean = this >= Int.MinValue && this <= Int.MaxValue
-           def isValidLong: Boolean = this >= Long.MinValue && this <= Long.MaxValue
+
+  override def isValidByte: Boolean = isValidLong && _long >= Byte.MinValue && _long <= Byte.MaxValue
+  override def isValidShort: Boolean = isValidLong && _long >= Short.MinValue && _long <= Short.MaxValue
+  override def isValidChar: Boolean = isValidLong && _long >= Char.MinValue && _long <= Char.MaxValue
+  override def isValidInt: Boolean = isValidLong && _long >= Int.MinValue && _long <= Int.MaxValue
```

Comment:
because `Long.MinValue` does not fit into any of these types, we can probably use `longEncoding` instead of `isValidLong` for these 4 methods

```suggestion
  override def isValidByte: Boolean  = longEncoding && _long >= Byte.MinValue && _long <= Byte.MaxValue
  override def isValidShort: Boolean = longEncoding && _long >= Short.MinValue && _long <= Short.MaxValue
  override def isValidChar: Boolean  = longEncoding && _long >= Char.MinValue && _long <= Char.MaxValue
  override def isValidInt: Boolean   = longEncoding && _long >= Int.MinValue && _long <= Int.MaxValue
```

Reply:
Done!